Assets/Scripts/Augment/GunFactory.cs

```diff
@@ -18,9 +18,56 @@ public static GameObject InstantiateGun(Item bodyPrefab, Item barrelPrefab, Item
         // Initialize everything
         gun.GetComponent<GunFactory>().InitializeGun(owner);
 
+        var cullingLayer = LayerMask.NameToLayer("Gun " + owner.inputManager.playerInput.playerIndex);
+        SetGunLayer(gun.GetComponent<GunFactory>(), cullingLayer);
+
+        GunFactory displayGun = owner.GunOrigin.GetComponent<GunFactory>();
+        displayGun.Body = bodyPrefab;
+        displayGun.Barrel = barrelPrefab;
+        displayGun.Extension = extensionPrefab;
+        displayGun.InitializeGun();
+
+        var cullingLayerDisplay = LayerMask.NameToLayer("Player " + owner.inputManager.playerInput.playerIndex);
+        SetGunLayer(displayGun, cullingLayerDisplay);
+
+        var firstPersonGunController = gun.GetComponent<GunFactory>().GunController;
+        var gunAnimations = displayGun.GetComponentsInChildren<AugmentAnimator>(includeInactive: true);
+        foreach (var animation in gunAnimations)
+        {
+            animation.OnInitialize(firstPersonGunController.stats);
+            firstPersonGunController.onFire += animation.OnFire;
+            firstPersonGunController.onReload += animation.OnReload;
+        }
+
+        firstPersonGunController.RightHandTarget = displayGun.GunController.RightHandTarget;
```

Comment:
It is not the same point. "e need to explictly say to the gunController which actually has a player refference (Display gun does not!) that the inverse kinematics for hands (Which is attached to players) should follow the display gun and NOT the first person gun. This is because we only want to calculate IK once and only for where it matters, a third person view.

The IK automatically handles orientation/position afterwards

Reply:
Oh okay, I forgor that we don't show right hands in first person :skull: